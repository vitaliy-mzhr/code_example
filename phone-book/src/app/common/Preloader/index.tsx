import * as React from 'react';
import { CircularProgress } from '@material-ui/core';
import PreloaderStyles from './styles';

const Preloader: React.StatelessComponent<{}> = () => (
  <PreloaderStyles>
    <CircularProgress/>
  </PreloaderStyles>
);

export default Preloader;
