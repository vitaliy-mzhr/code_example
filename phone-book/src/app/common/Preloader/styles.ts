import styled from 'styled-components';

const PreloaderStyles = styled.div`
  position: absolute;
  width: 100%;
  height: calc(100% - ${props => props.theme.dimensions.navMinHeightRem}rem);
  display: flex;
  justify-content: center;
  align-items: center;
  
  @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {
   height: calc(100% - ${props => props.theme.dimensions.navMinHeightRem * 0.9}rem);
  }
`;

export default PreloaderStyles;