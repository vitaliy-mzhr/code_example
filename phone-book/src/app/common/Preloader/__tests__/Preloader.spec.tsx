import * as React from 'react';
import { shallow } from 'enzyme';
import Preloader from '../../Preloader';

it('Preloader renders without crashing', () => {
  expect(shallow(<Preloader/>)).toMatchSnapshot();
});
