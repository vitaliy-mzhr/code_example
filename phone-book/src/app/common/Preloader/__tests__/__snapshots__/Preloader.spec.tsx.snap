// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader renders without crashing 1`] = `
<styled.div>
  <WithStyles(CircularProgress) />
</styled.div>
`;
